### ISSUE TITLE
Enable cataloging unlogged postgres tables  

### ISSUE THREAD

**Issue by ollivierj** [dbt-labs/dbt#3961]
### Describe the bug
We use **postgres 12.4.1** as a datawarehouse. During the ingestion processing, we use **unlogged tables** to speed up writing data into tables. Those tables are in a schema dedicated for storing raw data. We then transform this raw data using dbt.
  
We use the command below to generate the dbt documentation :
```
dbt docs generate --project-dir <MY_DBT_PROJECT> --profiles-dir <MY_PROFILE_DIR>
```

When looking at the dbt documentation, **none of the unlogged tables have their columns information filled up**.

We checked the dbt code source and found that **dbt does ignore unlogged and temporary tables when getting columns information** :
https://github.com/dbt-labs/dbt/blob/749f87397ec1e0a270b2e09bd8dbeb71862fdb81/plugins/postgres/dbt/include/postgres/macros/catalog.sql#L39    

### Steps To Reproduce
Create a postgres unlogged table, initialize a dbt project then run :
```
dbt docs generate
```

### Expected behavior
In the dbt documentation, i can see postgres unlogged table columns information 

### Screenshots and log output
![image](https://user-images.githubusercontent.com/8107615/135056818-ea4f684f-c946-4e91-92d7-15a662259794.png)

### System information
**Which database are you using dbt with?**
- [x ] postgres
- [ ] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
installed version: 0.20.2
   latest version: 0.20.2

Up to date!

Plugins:
  - bigquery: 0.20.2
  - snowflake: 0.20.2
  - postgres: 0.20.2
  - redshift: 0.20.2
```

**The operating system you're using:**
```
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
```

**The output of `python --version`:**
```
Python 3.8.8
```

**Comment by jtcohen6** [comment]
@ollivierj Good call! It sounds like the fix here is really quite minor, by changing the linked line to:
```sql
      and tbl.relpersistence in ('p', 'u') -- [p]ermanent table or [u]nlogged table. Exclude [t]emporary tables
```

It doesn't look like we have _any_ integration tests for the `unlogged` config, so now is as good a time as any to add one. I think this could be as simple as:
- Defining a model with `{{ config(materialized = 'table', unlogged=True) }}`
- Running it
- Executing a query (or dbt test!) that hits `pg_class` to confirm that the table is indeed unlogged
- Generating docs
- Confirming that the table appears in `catalog.json`

Would you be interested in contributing that fix? :)

**Comment by ollivierj** [comment]
@jtcohen6 Hi Jeremy,
Thank you for getting back to me. 
I will be available from December 11st for working on this issue.

**Comment by jtcohen6** [comment]
Sounds good! If someone else wishes to contribute the fix for this sooner, I'd welcome a PR for it.
